Error generating DataQualityPreset report with datetime column mapping and no reference dataset

I tried the DataQualityPreset test case with a datetime column mapping and no reference dataset.

The report failed with following error:
```
cls = <class 'evidently.metric_results.HistogramData'>, value = None

    @classmethod
    def from_df(cls, value: pd.DataFrame):
>       return cls(x=value["x"], count=value["count"])
E       TypeError: 'NoneType' object is not subscriptable
```

--- a/src/evidently/calculations/data_quality.py
+++ b/src/evidently/calculations/data_quality.py
@@ -377,7 +377,10 @@ def plot_data(
             reference = ref_data
             reference.columns = ["x", "count"]
         curr_data.columns = ["x", "count"]
-        data_hist = Histogram(current=HistogramData.from_df(curr_data), reference=HistogramData.from_df(reference))
+        data_hist = Histogram(
+            current=HistogramData.from_df(curr_data),
+            reference=HistogramData.from_df(reference) if reference is not None else None,
+        )
     elif column_type == ColumnType.Text:
         data_hist = None
     else:

--- a/tests/metric_preset/test_data_quality.py
+++ b/tests/metric_preset/test_data_quality.py
@@ -59,6 +59,26 @@ from evidently.report import Report
                 datetime="datetime",
             ),
         ),
+        (
+            pd.DataFrame(
+                {
+                    "myid": "some_id",
+                    "my_target": [1, 2, 3],
+                    "prediction": [1, 2, 3],
+                    "feature1": [1, 2, 3],
+                    "feature2": ["a", "b", "c"],
+                    "datetime": pd.date_range("2020-01-01", periods=3),
+                }
+            ),
+            None,
+            DataQualityPreset(),
+            ColumnMapping(
+                target="my_target",
+                id="myid",
+                prediction="prediction",
+                datetime="datetime",
+            ),
+        ),
     ),
 )
 def test_data_quality_preset(

